internal/cmd/request.go

```diff
@@ -0,0 +1,107 @@
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/api/server"
+	"github.com/edulinq/autograder/internal/common"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type CommonOptions struct {
+	Verbose bool `help:"Use verbose output to show full request/response without specific formatting." default:"false"`
+}
+
+func MustHandleCMDRequestAndExit(endpoint string, request any, responseType any) error {
+	return MustHandleCMDRequestAndExitFull(endpoint, request, responseType, CommonOptions{}, nil)
+}
+
+func MustHandleCMDRequestAndExitFull(endpoint string, request any, responseType any, options CommonOptions, customPrintFunc any) error {
```

Comment:
"Must" functions should not return an error.
They either work or panic/fatal/exit.

I see three options:
 1. Make these log.Fatal() on error.
 2. Make these non-Must functions that return an error.
 3. Do 2, but also make Must versions that call the non-must versions (like we do a lot in the db).

Since you also need to exit with a softer error (like where you exit with 2), that makes this a bit more complex.
Either you need to do 1, or add another return value ("success"?) to this function and do 3 (where the must function then understand the third return value).